pkg/kubectl/cmd/create_secret.go

```diff
@@ -0,0 +1,192 @@
+			cmd.Help()
+		},
+	}
+	cmd.AddCommand(NewCmdCreateSecretDockerRegistry(f, cmdOut))
+	cmd.AddCommand(NewCmdCreateSecretOpaque(f, cmdOut))
+	return cmd
+}
+
+const (
+	secretLong = `
+Create a secret based on a file, directory, or specified literal value
+
+A single secret may package one or more key/value pairs.
+
+When creating a secret based on a file, the key will default to the basename of the file, and the value will
+default to the file content.  If the basename is an invalid key, you may specify an alternate key.
+
+When creating a secret based on a directory, each file whose basename is a valid key in the directory will be
+packaged into the secret.  Any directory entries except regular files are ignored (e.g. subdirectories,
+symlinks, devices, pipes, etc).
+`
+
+	secretExample = `  // Create a new secret named my-secret with keys for each file in folder bar
+  $ kubectl create secret opaque my-secret --from-file=path/to/bar
+
+  // Create a new secret named my-secret with specified keys instead of names on disk
+  $ kubectl create secret opaque my-secret --from-file=ssh-privatekey=~/.ssh/id_rsa --from-file=ssh-publickey=~/.ssh/id_rsa.pub
+
+  // Create a new secret named my-secret with key1=supersecret and key2=topsecret
+  $ kubectl create secret opaque my-secret --from-literal=key1=supersecret --from-literal=key2=topsecret`
+)
+
+// NewCmdCreateSecretOpaque is a command to create generic secrets from files, directories, or literal values
+func NewCmdCreateSecretOpaque(f *cmdutil.Factory, cmdOut io.Writer) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "opaque NAME [--type=string] [--from-file=[key=]source] [--from-literal=key1=value1] [--dry-run=bool]",
+		Short:   "Create a secret from a local file, directory or literal value.",
+		Long:    secretLong,
+		Example: secretExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			err := CreateSecretOpaque(f, cmdOut, cmd, args)
+			cmdutil.CheckErr(err)
+		},
+	}
+	cmdutil.AddPrinterFlags(cmd)
+	cmd.Flags().String("generator", cmdutil.SecretV1GeneratorName, "The name of the API generator to use.")
+	cmd.Flags().StringSlice("from-file", []string{}, "Key files can be specified using their file path, in which case a default name will be given to them, or optionally with a name and file path, in which case the given name will be used.  Specifying a directory will iterate each named file in the directory that is a valid secret key.")
+	cmd.Flags().StringSlice("from-literal", []string{}, "Specify a key and literal value to insert in secret (i.e. mykey=somevalue)")
+	cmd.Flags().String("type", "", "The type of secret to create")
+	cmd.Flags().Bool("dry-run", false, "If true, only print the object that would be sent, without sending it.")
+	return cmd
+}
+
+// CreateSecretOpaque is the implementation the create secret opaque command
+func CreateSecretOpaque(f *cmdutil.Factory, cmdOut io.Writer, cmd *cobra.Command, args []string) error {
```

Comment:
That's not what this does.  It allows specifying a type, not necessarily opaque.